feat(account-profile): adding account-profile component

owilliams320 opened the pull request:
#Description
Adding account profile component. `VantageAccountProfileModule` will include two components

1. `vui-account-profile-menu` - menu content with ng content projectors to for `account-info-list` and `account-actions-list`
2. `vui-account-profile` - icon button and mat menu setup for `vui-account-profile-menu` component

closes #75
```
<vui-account-profile name="donald duck" email="donald.duck@teradata.com">
  <ng-container account-info-list>
    <mat-list-item></mat-list-item>
  </ng-container>
  <ng-container account-action-list>
    <button mat-list-item></button>
  </ng-container>
</vui-account-profile>
```

![account-menu](https://user-images.githubusercontent.com/3837706/79019773-172ce700-7b45-11ea-8397-c79c377921b6.gif)

christianmemije commented on src/docs-ui/app/app.component.html:
```diff
@@ -3,6 +3,38 @@
     <div td-toolbar-content layout="row" layout-align="start center" flex>
       <span flex></span>
 
+      <vui-account-profile name="donald duck" email="donald.duck@teradata.com">
```
I would call this component `vui-account-menu` or `vui-profile-menu` as long as there is menu in the name.

owilliams320 replied:
this component contains the button and the `vui-account-profile-menu` inside which can also menu used by itself

christianmemije replied:
Right, but when I read `vui-account-profile` I think a profile page. Might just be me though.

christianmemije commented on src/docs-ui/app/app.component.html:
```diff
@@ -3,6 +3,38 @@
+          </mat-list-item>
+        </ng-container>
+        <ng-container account-action-list>
+          <button
```
This theme toggling is going to be something that will be common. Can we make this re-useable? Either as in a re-usable component or maybe an input `themeToggle: boolean`?

owilliams320 replied:
yes makes sense to do that as a directive for the button. will do that as a separate effort not hold this PR up

christianmemije commented on src/docs-ui/app/app.component.html:
```diff
@@ -3,6 +3,38 @@
+            </button>
+          </ng-template>
+          <mat-divider></mat-divider>
+          <button mat-list-item>
```
Once again, sign out seems like it will be a common thing too. Actually it should always be in here. Since it is account, and signing in/out is the most essential thing regarding account.

owilliams320 replied:
the action is common yes but for Analyst we do not use the same auth service/method.

christianmemije replied:
Hmm maybe we can expose a logout output.

christianmemije commented on src/lib/account-profile/_account-profile-theme.scss:
```diff
@@ -0,0 +1,11 @@
+  vui-account-profile-menu {
+    .mat-list-base .mat-list-item .mat-line:nth-child(n + 2) {
+      color: $secondary-text;
+      @include mat-typography-level-to-styles($fontConfig, caption);
```
Hmm I can see how this might conflict with passed in markup, but would work for most cases...

christianmemije commented on src/lib/account-profile/account-profile-menu/account-profile-menu.component.html:
```diff
@@ -0,0 +1,15 @@
+<td-menu class="account-profile-menu">
+  <!--header-->
+  <mat-list td-menu-header>
+    <mat-list-item *ngIf="name" (click)="_blockEvent($event)">
```
What if no name but yes email?

owilliams320 replied:
accounted for this in my latest update. will make email top line style now if name is missing

christianmemije commented on src/lib/account-profile/account-profile-menu/account-profile-menu.component.spec.ts:
```diff
@@ -0,0 +1,45 @@
+  });
+
+  it('should block out click event on header click', () => {
+    const mouseEvent: MouseEvent = new MouseEvent("click",{bubbles: true, cancelable: true});
```
How about we test whether the menu stays open or not. Will require some changes to the fixture.

christianmemije commented on src/lib/account-profile/account-profile-menu/account-profile-menu.component.ts:
```diff
@@ -0,0 +1,17 @@
+  @Input() email: string;
+  @Input() name: string;
+
+  _blockEvent(event: Event): void {
```
private?

owilliams320 replied:
is used in the template to block the menu from hiding like in app switcher. cant be private if used in template

christianmemije replied:
Oops.

christianmemije commented on src/lib/account-profile/account-profile.component.html:
```diff
@@ -0,0 +1,10 @@
+  <mat-icon>person</mat-icon>
+</button>
+
+<mat-menu #accountMenu="matMenu" [overlapTrigger]="false">
```
What's `matMenu` used for?

owilliams320 replied:
this is containing component for the `account-profile-menu` and `button` that triggeers the mat menu. like the app switcher does

adamnel commented:
- [ ] Top two list items should be vertically centered in their 72px min-heights
- [ ] Add 8px below last item in the menu
- [ ] Add 8px below top list before the divider

christianmemije commented:
Adding more list items causes unaligned icons
![Screen Shot 2020-04-13 at 11 56 15 AM](https://user-images.githubusercontent.com/7193975/79153050-17fe8c80-7d82-11ea-8a1e-af6e5e6efd08.png)

owilliams320 commented:
@adamnel for the 8px already did the top you can see in the video. for bottom thats coming from `td-menu` need to update as a bug there

owilliams320 commented:
> Adding more list items causes unaligned icons
> ![Screen Shot 2020-04-13 at 11 56 15 AM](https://user-images.githubusercontent.com/7193975/79153050-17fe8c80-7d82-11ea-8a1e-af6e5e6efd08.png)

Looks like the comes from native Angular material. IMO should not mess with that

owilliams320 commented:
![Screen Shot 2020-04-13 at 4 13 59 PM](https://user-images.githubusercontent.com/3837706/79157301-615ed400-7da2-11ea-8ef8-ea40b6964444.png)

Addressed the UX comments. @christianmemije addressed most of your comments. the component is already reporting 100% test coverage :) . dont really want to fuss with adding more tests. i can see as a e2e test case IMO

christianmemije commented:
> the component is already reporting 100% test coverage :) 

Yes it might have 100% test coverage but that test does not seem too useful IMO.

owilliams320 commented:
@christianmemije looks like there is a desire to potentially have this in Covalent since it is a display only component can address that then
